### PR TITLE
[Subscriptions in Order Creation 3] Update signup fees by product quantity

### LINKED ISSUE #12564: [Subscriptions in Order Creation 3] Update sign-up fees when quantity > 1
Raised on p1714040401021719-slack-C025A8VV728

Sign-up fees are not multiplied by the subscription quantity, they're aded only once to the order per product. We can duplicate the price x quantity design here:

Screenshot 2024-04-25 at 19 09 26

This is computed on `CollapsibleProductRowCardViewModel.subscriptionConditionsSignupFee`, and rendered in `CollapsibleProductRowCard`View. We can use the same approach as with subscription prices:

```
        // Multiply price per unit by the currently selected quantity
        let quantity = stepperViewModel.quantity
        guard let decimalPrice = currencyFormatter.convertToDecimal(pricePerUnit)?.decimalValue,
              let stringTotal = currencyFormatter.formatHumanReadableAmount(decimalPrice * quantity) else {
            return nil
        }

        let formattedPrice = currencyFormatter.formatAmount(stringTotal)
        return formattedPrice
```

### PR DESCRIPTION
**Pull request by iamgabrielma**
Closes: #12564

## Description
This PR updates the product card UI to display signup fees correctly when a subscription item quantity is > 1. Up until now we would show only a single signup fee, but this needs to be multiplied by the item quantity:

Simulator Screen Recording - iPhone 15 - 2024-04-26 at 10 29 54

## Changes
- Update signup fees when quantity is updated
- Added a _"quantity x price"_ line to signup fees when quantity is > 1
- Extracted duplication when computing price * quantity into a helper function

## Testing instructions
* On a store using WooCommerce Subscriptions (you can use the testing site https://indiemelon.mystagingwebsite.com/)
* Tap `Orders` > `+` > `+ Add Product` > search for a subscription product with signup fees (named as "virtual sub" on my test store) > Add it to the order
* Expand the product card and observe that the signup fee line is updated when quantity changes, as well as the fee summary _(quantity x signup fee)_ appears and is updated when quantity > 1

Question: When we have only one subscription product with signup fees added to the order, I opted to not show the _"price x quantity"_ line unless quantity is > 1, since just adds unnecessary information to an already packed view. For consistency, I think we should also remove it from the product price line if quantity is 1,  so the view gets simpler. WDTY?

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by iamgabrielma** [comment]
Only one reviewer is needed. No rush as goes into 18.5 🙇

**Comment by bozidarsevo** [comment]
> so the view gets simpler

Yeah, that is good!

**Comment by bozidarsevo** [comment]
I see that signup fee is not included in order total?

Simulator Screenshot - iPhone 15 Pro - 2024-04-26 at 09 25 51

**Comment by joshheald** [comment]
> I see that signup fee is not included in order total?

Yeah, that's expected, it's pending a backend change to the Subscriptions extension.

Before we remove the feature flag I think we need to have warnings to merchants with too old a version of WooSubscriptions, which I think Jirka suggested. Alternatively we don't add support if the extension version is too old.

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/CollapsibleProductRowCardViewModel.swift [hunk 1]
```diff
@@ -356,6 +370,11 @@ private extension CollapsibleProductRowCardViewModel {
                 value: "%1$@ %2$@ free",
                 comment: "Description of the free trial conditions for a subscription product. " +
                 "Reads as: '3 days free'.")
+            static let signupFeeSummary = NSLocalizedString(
+                "CollapsibleProductRowCardViewModel.signupFeeSummary",
+                value: "%1$@ x %2$@",
+                comment: "Summary of quantity and signup fees for a subscription product when multiple are selected." +
+                "Reads as: '3 x $0.60'.")
```
Comment (joshheald) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/CollapsibleProductRowCardViewModel.swift]:
```suggestion
                value: "%1$@ × %2$@",
                comment: "Summary of quantity and signup fees for a subscription product when multiple are selected." +
                "Reads as: '3 × $0.60'. Please ensure you use a multiplication symbol, not a letter x")
```

File: WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/CollapsibleProductRowCardViewModelTests.swift [hunk 2]
```diff
@@ -542,13 +542,53 @@ final class CollapsibleProductRowCardViewModelTests: XCTestCase {
...
+        XCTAssertEqual(viewModel.subscriptionConditionsSignupFee, expectedSignUpFees)
+        XCTAssertEqual(viewModel.signupFeeSummary, expectedSignUpFeeSummary)
+    }
+
+    func test_productRow_when_subscription_has_signupFee_and_order_has_multiple_items_then_signupFees_are_multiplied_and_formatted() {
+        // Given
+        let signUpFee = "0.60"
+        let quantity = 10
+        let expectedSignUpFees = "$6.00"
+        let expectedSignUpFeeSummary = "10 x $6.00"
```
Comment (joshheald) [inline WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/CollapsibleProductRowCardViewModelTests.swift]:
```suggestion
        let expectedSignUpFeeSummary = "10 × $6.00"
```

**Review by joshheald — APPROVED** [review]
Looks good to me, and works well.

One thing I noticed – the vertical spacing between Price, Interval and Signup fee is inconsistent with designs... the price-interval gap (and others) looks too big, while the interval-signup fee gap looks too small.
### Actual
IMG_FC37CB8FC66D-1

### Designs
CleanShot 2024-04-26 at 15 57 58@2x

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/CollapsibleProductRowCardViewModel.swift (hunk 1 above)
Comment (iamgabrielma) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/CollapsibleProductRowCardViewModel.swift]: Thanks! Updated on https://github.com/woocommerce/woocommerce-ios/pull/12565/commits/029ab9d69d5fb57ef5cf90010f5696ce1e5b8bfd

File: WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/CollapsibleProductRowCardViewModelTests.swift (hunk 2 above)
Comment (iamgabrielma) [reply WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Creation/CollapsibleProductRowCardViewModelTests.swift]: https://github.com/woocommerce/woocommerce-ios/pull/12565/commits/029ab9d69d5fb57ef5cf90010f5696ce1e5b8bfd

**Comment by iamgabrielma** [comment]
> One thing I noticed – the vertical spacing between Price, Interval and Signup fee is inconsistent with designs... the price-interval gap (and others) looks too big, while the interval-signup fee gap looks too small.

Good catch, checking the figma designs it should be a 4 unit spacing between elements within the subscription details, and 16 units to the price.  I've updated the vertical spacing in subscription details here: https://github.com/woocommerce/woocommerce-ios/pull/12565/commits/a7bacbe041ff99c1da9869b88ca8033311a5a644 , however, the distance from the price to the rest seems to be correct so I left it as it is.

Screenshot 2024-04-30 at 17 43 28


>  feature flag

When re-testing this branch I've realized I've introduced a bug in the UI when moving the `Spacer` in this commit: https://github.com/woocommerce/woocommerce-ios/pull/12565/commits/8fdf886271338ac67a23d62d8143b5e60d6455ac . If the feature flag is set to false and the card is expanded for a non-subscription product, the price would appear pushed to the right of the screen, since we do not have subscription-related text to fill the content. I fixed this on https://github.com/woocommerce/woocommerce-ios/pull/12565/commits/c6a41f1530879d51aaed71b15cf71fe447a3d478

**Pull request merged**